src/_pytest/assertion/util.py

```diff
@@ -228,10 +230,32 @@ def _diff_text(left: str, right: str, verbose: int = 0) -> List[str]:
         explanation += ["Strings contain only whitespace, escaping them using repr()"]
     # "right" is the expected base against which we compare "left",
     # see https://github.com/pytest-dev/pytest/issues/3333
-    explanation += [
-        line.strip("\n")
-        for line in ndiff(right.splitlines(keepends), left.splitlines(keepends))
-    ]
+
+    from difflib import SequenceMatcher
+    stripped_left = ''.join(left.split())
+    stripped_right = ''.join(right.split())
+    s = SequenceMatcher(None, stripped_left, stripped_right)
+
+    nlines_left = left.count('\n')
+    nlines_right = right.count('\n')
+
+    if s.ratio() < 0.30 or max(nlines_left, nlines_right) < 5:
+        explanation += [
+            line.strip("\n")
+            for line in ndiff(right.splitlines(keepends), left.splitlines(keepends))
+        ]
+    else:
+        lines = ["=" * int((screen_width-18)/2) + " ACTUAL " + "=" * int((screen_width-18)/2)] 
```

Comment:
Can you please leave a comment explaining why `18` here? Same as `20` below (or even better assign them to self-explanatory variables if possible).

Reply:
I change code a little bit. I think it is more readable now.